lib/SYCL/SYCLArgsFlattening.cpp

```diff
@@ -433,6 +446,10 @@ static bool PrefixIn(const IndicesVector &Indices,
     // prefix exists.
     //
     // This load is safe if any prefix of its operands is safe to load.
+    for (auto itr = Indices.begin(); itr != Indices.end(); itr++)
+      DEBUG(dbgs() << "PrefixIn Indices: " << *itr << "\n");
+    for (auto itr = Low->begin(); itr != Low->end(); itr++)
+      DEBUG(dbgs() << "PrefixIn Low: " << *itr << "\n");
```

Comment:
There should be only 1  `DEBUG()` with the 2 loops inside.
Furthermore, why you do not use range-based for loop?